tests/tools/qe-data-observer/src/main/java/com/hazelcast/qe/Observer.java

```diff
@@ -0,0 +1,72 @@
+package com.hazelcast.qe;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import com.hazelcast.map.listener.MapListener;
+import com.hazelcast.qe.listener.AddEntryEventListener;
+import com.hazelcast.qe.listener.UpdateEntryEventListener;
+import com.hazelcast.qe.util.Locker;
+
+import java.util.concurrent.TimeUnit;
+
+public class Observer {
+    String mapName;
+    String observeOperation;
+    String clusterName;
+    Long timeout;
+    Integer eventsNumber;
+    Locker locker;
+    HazelcastInstance hzInstance;
+    TestResult testResult = new TestResult();
+
+    public Observer(String mapName, String observeOperation, String clusterName, Long timeout, Integer eventsNumber) {
+        this.mapName = mapName;
+        this.observeOperation = observeOperation;
+        this.clusterName = clusterName;
+        this.timeout = timeout;
+        this.eventsNumber = eventsNumber;
+    }
+
+    public void observe() throws Exception {
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.setClusterName(clusterName);
+
+        hzInstance = HazelcastClient.newHazelcastClient(clientConfig);
+        locker = new Locker();
+        IMap observableMap = hzInstance.getMap(mapName);
+        MapListener listener = getListener();
+        observableMap.addEntryListener(listener, true);
+        locker.lock(timeout, TimeUnit.MILLISECONDS);
```

Comment:
@ufukyilmaz we had a conversation with @olukas and, I believe, we need your help)
I thought that lock is a usual approach for wait-related tasks. At the same time, Ondrej mentioned that periodic check in the loop could simplify the code.
So let me ask you what approach you would choose? We agreed to take into account your opinion)

Reply:
When we use wait here, the test will definitely wait until a certain timeout, even if we make this observe very fast. That's why I think we shouldn't wait here anyway.

In addition, I didn't understand this code when I first read it either, I had to look inside this `Locker` code to understand what it is doing etc.. I think while introducing complex abstractions such as synchronization things (which are not always easy to understand), it should at least be well documented (I know it is also a relatively difficult task to document such things as they are complex.). For such reasons, I myself would look for something within the java concurrent library to do this instead of creating my own abstraction.

Reply:
@ufukyilmaz @olukas thank you for the comments, it has been completely redesigned - simplified per your coments